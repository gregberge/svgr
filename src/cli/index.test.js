/* globals jasmine */
import fs from 'mz/fs'
import { exec } from 'mz/child_process'

describe('cli', () => {
  beforeEach(() => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 10000
  })

  it('should work with a simple file', async () => {
    const [stdout] = await exec('bin/svgr __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('should work with stdin', async () => {
    const [stdout] = await exec('bin/svgr < __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--icon', async () => {
    const [stdout] = await exec('bin/svgr --icon __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--ids', async () => {
    const [stdout] = await exec('bin/svgr --ids __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--jsx-bracket-same-line', async () => {
    const [stdout] = await exec('bin/svgr --jsx-bracket-same-line __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--keep-useless-defs', async () => {
    const [stdout] = await exec('bin/svgr --keep-useless-defs __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--native', async () => {
    const [stdout] = await exec('bin/svgr --native __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-bracket-spacing', async () => {
    const [stdout] = await exec('bin/svgr --no-bracket-spacing __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-dimensions', async () => {
    const [stdout] = await exec('bin/svgr --no-dimensions __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-dimensions --no-view-box', async () => {
    const [stdout] = await exec('bin/svgr --no-dimensions --no-view-box __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-expand-props', async () => {
    const [stdout] = await exec(
      'bin/svgr --no-expand-props __fixtures__/one.svg',
    )
    expect(stdout).toMatchSnapshot()
  })

  it('--no-prettier', async () => {
    const [stdout] = await exec('bin/svgr --no-prettier __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-semi', async () => {
    const [stdout] = await exec('bin/svgr --no-semi __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-svgo', async () => {
    const [stdout] = await exec('bin/svgr --no-svgo __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-title', async () => {
    const [stdout] = await exec('bin/svgr --no-title __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--no-view-box', async () => {
    const [stdout] = await exec('bin/svgr --no-view-box __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--ref', async () => {
    const [stdout] = await exec('bin/svgr --ref __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--ref --no-expand-props', async () => {
    const [stdout] = await exec(
      'bin/svgr --ref --no-expand-props __fixtures__/one.svg',
    )
    expect(stdout).toMatchSnapshot()
  })

  it('--replace-attr-value', async () => {
    const [stdout] = await exec(
      'bin/svgr --replace-attr-value "#063855=currentColor" __fixtures__/one.svg',
    )
    expect(stdout).toMatchSnapshot()
  })

  it('--single-quote', async () => {
    const [stdout] = await exec('bin/svgr --single-quote __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--tab-width', async () => {
    const [stdout] = await exec('bin/svgr --tab-width 4 __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--template', async () => {
    const [stdout] = await exec(
      'bin/svgr --template __fixtures__/template.js __fixtures__/one.svg',
    )
    expect(stdout).toMatchSnapshot()
  })

  describe('--trailing-comma', () => {
    it('none', async () => {
      const [stdout] = await exec(`bin/svgr --trailing-comma none __fixtures__/one.svg`)
      expect(stdout).toMatchSnapshot()
    })

    it('es5', async () => {
      const [stdout] = await exec(`bin/svgr --trailing-comma es5 __fixtures__/one.svg`)
      expect(stdout).toMatchSnapshot()
    })

    it('all', async () => {
      const [stdout] = await exec(`bin/svgr --trailing-comma all __fixtures__/one.svg`)
      expect(stdout).toMatchSnapshot()
    })
  })

  it('--use-tabs', async () => {
    const [stdout] = await exec('bin/svgr --use-tabs __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
  })

  it('--out-dir', async () => {
    await exec('bin/svgr --out-dir __fixtures_build__ __fixtures__')
    expect(
      await fs.readFile('__fixtures_build__/One.js', 'utf-8'),
    ).toMatchSnapshot()
    expect(
      await fs.readFile('__fixtures_build__/nested/Two.js', 'utf-8'),
    ).toMatchSnapshot()
  })

  it('--help', async () => {
    const [stdout] = await exec('bin/svgr --help')
    const [shorthand] = await exec('bin/svgr -h')
    expect(stdout).toMatchSnapshot()
    expect(shorthand).toEqual(stdout);
  })

  it('--precision', async () => {
    const [stdout] = await exec('bin/svgr --precision 10 __fixtures__/one.svg')
    const [shorthand] = await exec('bin/svgr -p 10 __fixtures__/one.svg')
    expect(stdout).toMatchSnapshot()
    expect(shorthand).toEqual(stdout)
  })

  it('--version', async () => {
    const [stdout] = await exec('bin/svgr --version')
    const [shorthand] = await exec('bin/svgr -V')
    expect(stdout).toMatchSnapshot()
    expect(shorthand).toEqual(stdout)
  })
})
