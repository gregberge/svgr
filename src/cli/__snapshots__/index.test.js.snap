// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cli --icon 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width=\\"1em\\" height=\\"1em\\" viewBox=\\"0 0 48 1\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --ids 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <g id=\\"Page-1\\" fill=\\"none\\" fillRule=\\"evenodd\\">
      <g id=\\"19-Separator\\" transform=\\"translate(-129 -156)\\" fill=\\"#063855\\">
        <g id=\\"Controls/Settings\\" transform=\\"translate(80)\\">
          <g id=\\"Content\\" transform=\\"translate(0 64)\\">
            <g id=\\"Group\\" transform=\\"translate(24 56)\\">
              <g id=\\"Group-2\\">
                <path id=\\"Rectangle-5\\" d=\\"M25 36h48v1H25z\\" />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);

export default One;

"
`;

exports[`cli --no-expand-props 1`] = `
"import React from \\"react\\";

const One = () => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\">
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --no-prettier 1`] = `
"import React from 'react'

const One = (props) => <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
  <title>
    Rectangle 5
  </title>
  <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
</svg>


export default One
"
`;

exports[`cli --no-semi 1`] = `
"import React from \\"react\\"

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
)

export default One

"
`;

exports[`cli --no-svgo 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg
    width=\\"48px\\"
    height=\\"1px\\"
    viewBox=\\"0 0 48 1\\"
    version={1.1}
    xmlnsXlink=\\"http://www.w3.org/1999/xlink\\"
    {...props}
  >
    <title>Rectangle 5</title>
    <desc>Created with Sketch.</desc>
    <defs />
    <g id=\\"Page-1\\" stroke=\\"none\\" strokeWidth={1} fill=\\"none\\" fillRule=\\"evenodd\\">
      <g
        id=\\"19-Separator\\"
        transform=\\"translate(-129.000000, -156.000000)\\"
        fill=\\"#063855\\"
      >
        <g id=\\"Controls/Settings\\" transform=\\"translate(80.000000, 0.000000)\\">
          <g id=\\"Content\\" transform=\\"translate(0.000000, 64.000000)\\">
            <g id=\\"Group\\" transform=\\"translate(24.000000, 56.000000)\\">
              <g id=\\"Group-2\\">
                <rect id=\\"Rectangle-5\\" x={25} y={36} width={48} height={1} />
              </g>
            </g>
          </g>
        </g>
      </g>
    </g>
  </svg>
);

export default One;

"
`;

exports[`cli --no-title 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --no-view-box 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --precision 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --replace-attr-value 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"currentColor\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli --template 1`] = `
"import React from \\"react\\";

export default function One() {
  return (
    <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
      <title>Rectangle 5</title>
      <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
    </svg>
  );
}

"
`;

exports[`cli should work with a simple file 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;

"
`;

exports[`cli should work with output directory 1`] = `
"import React from \\"react\\";

const One = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default One;
"
`;

exports[`cli should work with output directory 2`] = `
"import React from \\"react\\";

const Two = props => (
  <svg width={48} height={11} viewBox=\\"0 0 48 11\\" {...props}>
    <title>Group 6</title>
    <g stroke=\\"#063855\\" fill=\\"none\\" fillRule=\\"evenodd\\">
      <path d=\\"M0 7l5.317-2.658a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0L48 7\\" />
      <path d=\\"M0 10l5.317-2.658a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0L48 10M0 4l5.317-2.658a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0l2.634 1.316a6 6 0 0 0 5.366 0l2.634-1.316a6 6 0 0 1 5.366 0L48 4\\" />
    </g>
  </svg>
);

export default Two;
"
`;

exports[`cli should work with stdin 1`] = `
"import React from \\"react\\";

const SvgComponent = props => (
  <svg width={48} height={1} viewBox=\\"0 0 48 1\\" {...props}>
    <title>Rectangle 5</title>
    <path d=\\"M0 0h48v1H0z\\" fill=\\"#063855\\" fillRule=\\"evenodd\\" />
  </svg>
);

export default SvgComponent;

"
`;
